pkg/verify/tlog_test.go

```diff
@@ -221,3 +228,68 @@ func TestMaxAllowedTlogEntries(t *testing.T) {
 	_, err = verify.VerifyArtifactTransparencyLog(&tooManyTlogEntriesEntity{entity}, virtualSigstore, 1, true, false)
 	assert.ErrorContains(t, err, "too many tlog entries") // too many tlog entries should fail to verify
 }
+
+type mockEntriesClient struct {
+	Entries []*models.LogEntry
+}
+
+func (m *mockEntriesClient) CreateLogEntry(_ *entries.CreateLogEntryParams, _ ...entries.ClientOption) (*entries.CreateLogEntryCreated, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) GetLogEntryByIndex(params *entries.GetLogEntryByIndexParams, _ ...entries.ClientOption) (*entries.GetLogEntryByIndexOK, error) {
+	resp := &entries.GetLogEntryByIndexOK{}
+	if len(m.Entries) != 0 {
+		for _, e := range m.Entries {
+			for _, i := range *e {
+				if *i.LogIndex == params.LogIndex {
+					resp.Payload = *e
+				}
+			}
+		}
+
+		if resp.Payload == nil {
+			resp.Payload = *m.Entries[0]
+		}
+	}
+	return resp, nil
+}
+
+func (m *mockEntriesClient) GetLogEntryByUUID(_ *entries.GetLogEntryByUUIDParams, _ ...entries.ClientOption) (*entries.GetLogEntryByUUIDOK, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) SearchLogQuery(_ *entries.SearchLogQueryParams, _ ...entries.ClientOption) (*entries.SearchLogQueryOK, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) SetTransport(_ runtime.ClientTransport) {}
+
+func TestOnlineVerification(t *testing.T) {
+	virtualSigstore, err := ca.NewVirtualSigstore()
+	assert.NoError(t, err)
+
+	statement := []byte(`{"_type":"https://in-toto.io/Statement/v0.1","predicateType":"customFoo","subject":[{"name":"subject","digest":{"sha256":"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef"}}],"predicate":{}}`)
+	entity, err := virtualSigstore.AttestWithInclusionProof("foo@example.com", "issuer", statement)
+	assert.NoError(t, err)
+	tlogEntries, err := entity.TlogEntries()
+	assert.NoError(t, err)
+
+	tlogEntry := tlogEntries[0]
+	var logEntry models.LogEntry = make(models.LogEntry)
+	logEntry["foo"] = models.LogEntryAnon{
+		Body:           tlogEntry.Body(),
+		IntegratedTime: swag.Int64(tlogEntry.IntegratedTime().Unix()),
+		LogIndex:       swag.Int64(tlogEntry.LogIndex()),
+		LogID:          swag.String(hex.EncodeToString([]byte(tlogEntry.LogKeyID()))),
+		Verification:   tlogEntry.Verification(),
```

Comment:
Looking closer at the functional changes in entry.go that were needed to make this work, I'm seeing that this test is tautological - it's using the entity returned by the attestation as the response to the mock rekor client, and effectively comparing it to itself. I think this can be avoided, and the changes in entry.go can be discarded, if we create the mocked response from scratch with some added helpers to VirtualSigstore: https://gist.github.com/cmurphy/44a742f9f08e5cb56bc79dccfbd3b327

Reply:
@cmurphy that makes sense, thank you so much for helping with this!